dvc/stage/cache.py

```diff
@@ -197,7 +198,7 @@ def _transfer(func, from_remote, to_remote):
             return []
 
         for src in from_remote.walk_files(runs):
-            rel = src.relative_to(from_remote.path_info)
+            rel = Path(os.path.relpath(src, from_remote.path_info))
```

Comment:
```suggestion
            rel = PathInfo(os.path.relpath(src, from_remote.path_info))
```

DVC `PathInfo` inherits from python's `PurePath` and should be preferred here